extension/lib/addProductsToCart.js

```diff
@@ -25,19 +24,10 @@ module.exports = async (context, input) => {
 
   const shopgateCartPipeline = ShopgateCartPipeline.create(context)
   try {
-    await shopgateCartPipeline.addProducts(input.products).catch((err) => {
-      if (err.code === 422) {
-        const message = new ShopgateCartMessage('error', 'This product is not available anymore', 422)
-        return {
-          cartId: '',
-          messages: [message.toJson()]
-        }
-      }
-      return {}
-    })
-  } catch (error) {
-    context.log.error(decorateError(error), 'Failed adding products to cart')
-    throw new Error()
+    await shopgateCartPipeline.addProducts(input.products)
+  } catch (err) {
+    context.log.error(decorateError(err), 'Failed adding products to cart')
```

Comment:
this is not needed anymore, this is logged already! You also need to clean up UpdateProductsToCart as well!